src/metabase/integrations/slack.clj

```diff
@@ -135,84 +144,96 @@
 
 (defn conversations-list
-  "Calls Slack API `conversations.list` and returns list of available 'conversations' (channels and direct messages). By
-  default only fetches channels."
+  "Calls Slack API `conversations.list` and returns list of available 'conversations' (channels and direct messages).
+  By default only fetches channels, and returns them with their # prefix. Note the call to [[paged-list-request]] will
+  only fetch the first [[max-list-results]] items."
   [& {:as query-parameters}]
   (let [params (merge {:exclude_archived true, :types "public_channel"} query-parameters)]
-    (paged-list-request "conversations.list" :channels params)))
+    (paged-list-request "conversations.list"
+                        ;; response -> channel names
+                        #(->> %
+                              :channels
+                              (map (fn [channel]
+                                     (str \# (:name channel)))))
+                        params)))
 
-(defn channel-with-name
-  "Return a Slack channel with `channel-name` (as a map) if it exists."
+(defn channel-exists?
+  "Returns true if the channel it exists."
   [channel-name]
-  (some (fn [channel]
-          (when (= (:name channel) channel-name)
-            channel))
-        (conversations-list)))
+  (and channel-name
+       (contains?
+        (set (slack-cached-channels-and-usernames))
+        (str \# channel-name))))
 
 (s/defn valid-token?
-  "Check whether a Slack token is valid by checking whether we can call `conversations.list` with it."
+  "Check whether a Slack token is valid by checking if the `conversations.list` Slack api accepts it."
   [token :- su/NonBlankString]
   (try
     (binding [*send-token-error-emails?* false]
-      (boolean (take 1 (conversations-list :limit 1, :token token))))
+      (boolean (take 1 (:channels (GET "conversations.list" :limit 1, :token token)))))
     (catch Throwable e
       (if (slack-token-error-codes (:error-code (ex-data e)))
         false
         (throw e)))))
 
 (defn users-list
-  "Calls Slack API `users.list` endpoint and returns the list of available users."
+  "Calls Slack API `users.list` endpoint and returns the list of available users with their @ prefix. Note the call
+  to [[paged-list-request]] will only fetch the first [[max-list-results]] items."
   [& {:as query-parameters}]
-  (->> (paged-list-request "users.list" :members query-parameters)
+  (->> (paged-list-request "users.list"
+                           ;; response -> user names
+                           #(->> %
+                                 :members
+                                 (map (fn [member]
+                                        (str \@ (:name member)))))
+                           query-parameters)
        ;; filter out deleted users and bots. At the time of this writing there's no way to do this in the Slack API
        ;; itself so we need to do it after the fact.
        (filter (complement :deleted))
```

Comment:
I wouldn't bother changing this now but for next time you can use `remove` e.g. `(remove :deleted)`. `remove` is basically the same as `filter` + `complement`

Reply:
Oh nvm I'm dumb this isn't even part of your PR. Forget I said anything

Reply:
I fixed it in a subsequent commit anyway